### PR TITLE
ENH: add nan_policy to `stats.kendalltau`.

### LINKED ISSUE #628: Statistics Review: kendalltau (Trac #101)
_Original ticket http://projects.scipy.org/scipy/ticket/101 on 2006-04-02 by @rkern, assigned to @rkern._

The function `kendalltau` in file source:trunk/Lib/stats/stats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### PR DESCRIPTION
**Pull request by aeklant**
closes gh-628

### CODE REVIEW

**Comment by aeklant** [comment]
@rgommers ping.

Added:
* `nan_policy`
* Explicit handling of unequally sized input
* See also section

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -3445,9 +3454,23 @@ def kendalltau(x, y, initial_lexsort=True):
 
     KendalltauResult = namedtuple('KendalltauResult', ('correlation', 'pvalue'))
 
-    if not x.size or not y.size:
+    if x.size != y.size:
+        raise ValueError('All inputs must be of the same size')
```
Comment (rgommers) [inline scipy/stats/stats.py]: Maybe help the user a bit more here:

    All inputs to `kendalltau` must be of the same size,
    found x-size %s and y-size %s" % (x.size, y.size)"

**Comment by rgommers** [comment]
One minor suggestion, everything else looks good to me.

**Comment by aeklant** [comment]
Comment addressed.

**Comment by rgommers** [comment]
LGTM, merged. Thanks @aeklant

**Pull request merged**